tinygrad/codegen/uops.py

```diff
@@ -405,7 +405,16 @@ def push(u):
     self._uops = self._uops[:-1]
 
     # TODO: ifs should be removed and just the store should be gated
-    for u in ifs[::-1]: self._uops.append(UOp(UOps.ENDIF, None, (u,)))
+    dfs_until_bar_cache = {}
+    def dfs_until_bar(u: UOp):
```

Comment:
could this be simpler now that we don't barrier stores?

Reply:
I don't this has to do with barrier-ing stores, but it can be worked into the existing code